quarto: add page (#8254)
tldr-pages/tldr: pages/common/quarto.md

# quarto

> An open-source scientific and technical publishing system built on Pandoc.
> More information: <https://quarto.org/>.

- Create a Quarto project:

> `quarto create-project {{destination_directory}}`

Reviewer: ```suggestion
`quarto create-project {{path/to/project}}`
```
This also applies to the 2nd and 3rd example.

- Create a new website project:

> `quarto create-project {{destination_directory}} --type {{website}}`

Reviewer 2: ```suggestion
`quarto create-project {{path/to/destination_directory}} --type {{website}}`
```

Reviewer: Whoops, I thought this was already fixed 😅

> `quarto create-project {{destination_directory}} --type website`

Reviewer: ```suggestion
`quarto create-project {{destination_directory}} --type {{website}}`
```
Even tough it is a specific example, it still is an argument. This applies to all `--type` and `--to` flags.

- Create a new book project:

> `quarto create-project {{destination_directory}} --type {{book}}`

Reviewer 2: ```suggestion
`quarto create-project {{path/to/destination_directory}} --type {{book}}`
```

> - Render R Markdown file to HTML:

Reviewer: ```suggestion
- Render an R Markdown file to HTML:
```

> `quarto render {{input.rmd}} --to html`

Reviewer: ```suggestion
`quarto render {{path/to/file.rmd}} --to html`
```
Same for the other examples.

> - Render Quarto file to HTML:

Reviewer: ```suggestion
- Render a Quarto file to HTML:
```
We should always write complete sentences and not leave out articles.

`quarto render {{input.qmd}} --to html`

- Render Jupyter notebooks to HTML:

`quarto render {{input.ipynb}} --to html`

> - Render document to PDF:

Reviewer: Which kind of document is this? The command itself is exactly the same as rendering a Quarto file.

Author: It is a Jupyter Notebook.

Reviewer: Sorry, wrong line. I mean the `input.qmd` file.

Author: It is a Quarto document. Very similar to an R Markdown document: https://quarto.org/docs/get-started/hello/rstudio.html

Reviewer: Does converting to `docx` and `pdf` only work for Quarto documents?

> - Create a Quarto project:
> 
> `quarto create-project {{path/to/project}}`
> 
> - Create a new website project:
> 
> `quarto create-project **{{destination_directory}}** --type {{website}}`
> 
> - Create a new book project:
> 
> `quarto create-project **{{destination_directory}}** --type {{book}}`
> 
> - Render an R Markdown file to HTML:
> 
> `quarto render {{path/to/file.rmd}} --to html`
> 
> - Render a Quarto file to HTML:
> 
> `quarto render {{path/to/file.qmd}} --to html`
> 
> - Render a Jupyter notebook to HTML:
> 
> `quarto render {{path/to/file.ipynb}} --to html`
> 
> - Render a document to PDF:
> 
> `quarto render {{path/to/file.qmd}} --to pdf`
> 
> - Render a document to Microsoft docx document:
> 
> `quarto render {{path/to/file.qmd}} --to docx`

Reviewer: ```suggestion
- Create a Quarto project:

`quarto create-project {{path/to/project}}`

- Create a new website project:

`quarto create-project {{path/to/project}} --type {{website}}`

- Create a new book project:

`quarto create-project {{path/to/project}} --type {{book}}`

- Render an R Markdown file to HTML:

`quarto render {{path/to/file.rmd}} --to {{html}}`

- Render a Quarto file to HTML:

`quarto render {{path/to/file.qmd}} --to {{html}}`

- Render a Jupyter notebook to HTML:

`quarto render {{path/to/file.ipynb}} --to {{html}}`

- Render a document to PDF:

`quarto render {{path/to/file.qmd}} --to {{pdf}}`

- Render a document to Microsoft docx document:

`quarto render {{path/to/file.qmd}} --to {{docx}}`
```

Author: I think I managed to fix it. Do not hesitate if I missed something.

Reviewer: Well, you didn't apply this suggestion. The argument after `--too` as to be in `{{ }}` and it should be `{{path/to/project}}` instead of `**{{destination_directory}}**`.

> - Render a Quarto file to PDF:
> 
> `quarto render {{path/to/file.qmd}} --to {{pdf}}`
> 
> - Render a Quarto file to Microsoft docx document:
> 
> `quarto render {{path/to/file.qmd}} --to {{docx}}`

Reviewer: Otherwise it sounds like this only works for `.qmd` files.
```suggestion
- Render an R Markdown, Jupyter, or Quarto file to PDF:

`quarto render {{path/to/file}} --to {{pdf}}`

- Render an R Markdown, Jupyter, or Quarto file to a Microsoft `docx` document:

`quarto render {{path/to/file}} --to {{docx}}`
```
